src/client/builtin_scene/web_content_renderer.cpp

```diff
@@ -1082,9 +1115,22 @@ namespace builtin_scene::web_renderer
     auto webContentMaterial = material3d->material<materials::WebContentInstancedMaterial>();
     if (webContentMaterial)
     {
-      // Check if this is a spatial image and set the spatial flag
-      auto imageComponent = getComponent<Image2d>(entity);
-      // Use the same texture update method for both spatial and non-spatial images
+      // Set SDF flag on instance for text entities
+      auto textComponent = getComponent<Text2d>(entity);
+      if (textComponent)
+      {
+        // Get the instance and mark it for SDF texture usage
+        auto &instance = material3d->getInstance(entity);
+        instance.setUseSDFTexture(true);
+      }
+      else
+      {
+        // For non-text entities, ensure SDF is disabled
+        auto &instance = material3d->getInstance(entity);
+        instance.setUseSDFTexture(false);
```

Comment:
You need to update this state of the instance at `RenderSystem::updateInstancedMeshData` which reads the state from `WebContent` to the instance data.